packages/now-client/src/deploy.ts

```diff
@@ -59,18 +64,31 @@ async function* createDeployment(
       };
     }
 
+    const regex = /^x-(?:vercel|now)-(?:warning|notice|tip)-(.*)$/;
```

Comment:
I think you can utilize this regex to group match and remove all the `startsWith()` checks below.